modules/cli/src/main/scala/scala/cli/commands/shared/SharedOptions.scala

```diff
@@ -330,7 +330,11 @@ final case class SharedOptions(
         platform = platformOpt.map(o => Positioned(List(Position.CommandLine()), o))
       ),
       scriptOptions = bo.ScriptOptions(
-        codeWrapper = None
+        codeWrapper = platformOpt match {
+          case Some(_: Platform.JS.type)                    => Some(ObjectCodeWrapper)
+          case _ if scalaVersion.exists(_.startsWith("2.")) => Some(ObjectCodeWrapper)
```

Comment:
@tgodzik Here, when creating BuildOptions

Reply:
thanks!

Reply:
@MaciejG604 It seems this will only work if the Scala version is specified on the command-line, right? What if it's specified via a using directive in a script?

Reply:
Should work for both using directives and cli options, wrapper inference is now postponed to the moment after preprocessing, when build options are extracted from both inputs and cli options.